spec/spec_helper.rb

```diff
@@ -31,9 +31,11 @@
     rspec.syntax = :expect
   end
 
-  c.before do
-    allow(ENV).to receive(:fetch).with("OPENAI_ACCESS_TOKEN", nil).and_return("abcdef") unless ENV["OPENAI_ACCESS_TOKEN"]
-    allow(ENV).to receive(:fetch).with("SERPAPI_API_KEY", nil).and_return("abcdef") unless ENV["SERPAPI_API_KEY"]
+  c.before do |example|
+    token = example.metadata[:skip_tokens] ? nil : "abcdef"
+    allow(ENV).to receive(:fetch).with("OPENAI_ACCESS_TOKEN", nil).and_return(token)
+    allow(ENV).to receive(:fetch).with("SERPAPI_API_KEY", nil).and_return(token)
```

Comment:
When I run the tests in my environment, I want the keys to come from my ENV so that I can generate the cassettes. I think this will ignore that right?

Reply:
Maybe just instead of "adbdef" you could do: `ENV.fetch(key, "abcdef")`, but you would need to do this twice.

Reply:
I thought about it, but doing this might make it more complicated to contribute to the gem.

For example, to help developing things or playing with the gem on my own, I have to create .env file with my own creds. As soon as I do it all the tests start failing due to new web request going to the servers and VCR can't handle it.

That's exactly the problem that I faced so I had to make these changes.

Since for testing we use VCR and you probably will need to update VCR only once in a while, it's easier to put real keys instead of `and_return(token)` once temporally and update VCR.

Reply:
I'm not sure that I follow. The cassettes take precedence. If you want to make a change, you just delete a cassette and it regenerates (assuming you have the keys in your .env file). Otherwise, your keys are never used.

Reply:
That's not exactly true, if you change the key, your API request changes and VCR no longer can detect it, so it throws an error that you are sending an "unrecognized" request and fails the test.

```
     Failure/Error:                                                                                                                                                                                        
       expect do                                                                                                                                                                                           
         described_class.new.client(prompt: "write a poem")                                                                                                                                                
       end.to raise_error(Boxcars::ConfigurationError)                                                                                                                                                     
                                                                                                                                                                                                           
       expected Boxcars::ConfigurationError, got #<VCR::Errors::UnhandledHTTPRequestError:                                                                                                                 
                                                                                                                                                                                                           
       =========================================================...n/ignore-request                                                                                                                        
       ================================================================================     
```

You can test it by changing your env key to the different one and running `rspec ./spec/boxcars/openai_spec.rb:5` on main branch